src/main/java/chess/domain/board/Board.java

```diff
@@ -11,22 +12,40 @@
 
 public class Board {
     private final Map<Position, Piece> piecePosition;
+    private final Color turn;
 
-    public Board(Map<Position, Piece> pieces) {
+    public Board(Map<Position, Piece> pieces, Color color) {
         piecePosition = new HashMap<>(pieces);
+        turn = color;
     }
 
-    public void movePiece(Position origin, Position destination) {
+    public Board movePiece(Position origin, Position destination) {
         validateMoveRequest(origin, destination);
         Piece targetPiece = piecePosition.get(origin);
         if (!targetPiece.canMove(origin, destination, piecePosition.get(destination))) {
             throw new IllegalPieceMoveException();
         }
-        piecePosition.put(destination, targetPiece);
-        piecePosition.put(origin, EmptyPiece.getInstance());
+        return new Board(movedBoard(origin, destination), nextTune());
+    }
+
+    private Map<Position, Piece> movedBoard(Position origin, Position destination) {
+        HashMap<Position, Piece> movedBoard = new HashMap<>(piecePosition);
+        movedBoard.put(destination, piecePosition.get(origin).nextPiece());
+        movedBoard.put(origin, EmptyPiece.getInstance());
+        return movedBoard;
+    }
+
+    private Color nextTune() {
```

Comment:
```suggestion
    private Color nextTurn() {
```

매의 눈 @_@
Color.NONE이 설정이 된다면 올바르지 않게 동작할 수 있겠네요!

Reply:
Board에서 필드를 초기화하기 전 Color가 None이 들어오면 생성되지 않도록 바꿨습니다!